### ISSUE TITLE
Databinding characteristics are hard to read/maintain

### ISSUE THREAD

**Issue by ebabel** [ponewheel/android-ponewheel#21]
This is hard to figure out whats going on:
android:text='@{owdevice.deviceReadCharacteristics.get(4).ui_name}'

**Comment by kwatkins** [comment]
Yeah - this one is confusing and never turned out like I wanted. The intent was to be able to easily remove/add deviceReadCharacteristics and that be the driver for the ui (ui_name, flags like ui_enabled etc) via data binding. If I recall I think it hit some limits with data binding? or didn't understand some part of it. I think it had to something with limitations of iterations in data binding, or some shit like that. 

I'm all for whatever you think should be done to make it clear. The layout_row was some grouping thing I was trying (notifying BLE chars vs read-once values), agree just putting them in order would clear that confusion up.

**Comment by ebabel** [comment]
Hmm, maybe abandon databinding? Its definitely nice to have, but also has its limits. I bet we could maybe come up with a dynamic solution that iterates over the characteristics and creates views programmatically. Not sure if its worth the effort but could be cleaner and more powerful.

**Comment by kwatkins** [comment]
Would take a bit of work for little return right now, agree we should avoid it when we can and slowly phase it out.
